src/frontend/screens/AuthScreen.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { View, Image, Text, StyleSheet, BackHandler } from "react-native";
+
+import { DARK_BLUE, RED, WHITE } from "../lib/styles";
+import { SecurityContext } from "../context/SecurityContext";
+import { PasscodeInput } from "../sharedComponents/PasscodeInput";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { AppStackList } from "../Navigation/AppStack";
+import { useFocusEffect } from "@react-navigation/native";
+
+const m = defineMessages({
+  enterPass: {
+    id: "screens.EnterPassword.enterPass",
+    defaultMessage: "Enter your passcode",
+  },
+  wrongPass: {
+    id: "screens.EnterPassword.wrongPass",
+    defaultMessage: "Incorrect passcode, please try again ",
+  },
+});
+
+export const AuthScreen = ({
+  navigation,
+}: NativeStackScreenProps<AppStackList, "AuthScreen">) => {
+  const [error, setError] = React.useState(false);
+  const { authenticate, authState } = React.useContext(SecurityContext);
+  const [inputtedPass, setInputtedPass] = React.useState("");
+
+  useFocusEffect(
+    React.useCallback(() => {
+      const subscription = BackHandler.addEventListener(
+        "hardwareBackPress",
+        () => {
+          // This is not being fired?!
+          return true;
+        }
+      );
+      console.log(subscription);
+      return () => {
+        if (true) {
+          subscription.remove();
+        }
+      };
+    }, [])
+  );
```

Comment:
Why are you wrapping in a useCallback? I think that is breaking things, e.g. you are returning within the useCallback, not the useFocusEffect, and why the `if (true)`?

Reply:
@gmaclennan that's unfortunately the usage that's presented in the react-native-navigation docs. It's a pretty weird api

https://reactnavigation.org/docs/use-focus-effect/
https://reactnavigation.org/docs/custom-android-back-button-handling/

Reply:
Ah, I see you are copying from the React Native docs. I'm not sure why they are implementing it like that? Have you also seen https://reactnavigation.org/docs/preventing-going-back/ ? What are you testing also? This will only override the hardware back button, not other ways of going back.

Reply:
yeah we didn't look at that, that may be what's needed!

Reply:
im only trying to prevent the hardware back button due (there is no header) AND the other approach is causing problems. The password is authenticated, and we try to go back() into the app, but the back is being prevented because the goBack() is being called before the next render

Reply:
If trying the other approach, then you need a check for `authState === 'authenticated'` in the `beforeRemove` handler, to bail early before calling `e.preventDefault`.

Reply:
I have tried that. The problem is that `navigate.goBack()` and `setAuthState(authenticated)` are happening in the same function. Essentially, `goBack()` is always being called before the authState has changed. I can pass the remove listener function to the authenticate function though...kind of round about, but should work

Reply:
I tried that initially. The problem is that navigation is only accessible in the `appContainerWrapper` and not in the security context. This mean, the `goBack` had to be handled in this wrapper. So having a `useEffect` that fired `goBack()` when the `authState === 'authenticated` has the undesired effect of `goBack()` being fired when the user was not using a passcode (because when they are not using a passcode, they are `authenticated`).